core/src/main/scala/kafka/server/KafkaHealthcheck.scala

```diff
@@ -73,6 +81,8 @@ class KafkaHealthcheck(private val brokerId: Int,
     @throws(classOf[Exception])
     def handleStateChanged(state: KeeperState) {
       // do nothing, since zkclient will do reconnect for us.
+      if(state == KeeperState.Expired)
+        shutdown()
```

Comment:
Do we need to do the same (i.e. clean-up zkWatchedEphemeral) for other handleStateChanged as well?